cpp/include/raft/sparse/solver/detail/mst_solver_inl.cuh

```diff
@@ -210,7 +210,9 @@ alteration_t MST_solver<vertex_t, edge_t, weight_t, alteration_t>::alteration_ma
   auto init  = tmp.element(1, stream) - tmp.element(0, stream);
   auto max   = thrust::transform_reduce(
     policy, begin, end, alteration_functor<weight_t>(), init, thrust::minimum<weight_t>());
-  return max / static_cast<alteration_t>(2);
+  // Enforce distinct weights if initial edge weights are identical by returning
+  // a value of 1
+  return max > 0 ? max / static_cast<alteration_t>(2) : 1;
```

Comment:
@jnke2016 Consider the case where all edge weights are `0.1`, and not `0.1 | 0.2`. Here, I think, the tests would fail - as code is assuming that if alternation_max is `0` (all weights are the same), they must be `1`. For this case, you could pick `max` as any edge weight - which here should be `0.1`.

Reply:
@wolfram77 I just pushed a test with all edge weights being 0.1 and it passes.

Reply:
> hey must be 1. For this case, you could pick max as any edge weight - which here should be 0.1. For this case, you could pick max as any edge weight - which here should be 0.1

That value doesn't really matter. Only the relatively order of the edges does where in this cases there is none.

Reply:
@jnke2016 Oh yeah, you are right, my bad.

Reply:
no worries @wolfram77, if you have any other concerns please feel free to reach out.